src/client/components/three/three.tsx

```diff
@@ -54,3 +63,45 @@ export class Three extends Component<{ stage(canvas: Canvas): IComputedValue<Sta
   }
 }
 
+export class StageCache {
+  private readonly camerasPerType = new Map<string, Camera>()
+  latestCamera?: Camera
+  readonly scene = new Scene()
+
+  get camera(): Camera {
+    if (!this.latestCamera) {
+      throw new Error('Precondition: No camera available')
+    }
+    return this.latestCamera
+  }
+
+  copy(stage: Stage): Stage {
+    this.copyToLocalScene(stage.scene)
+    this.copyToLocalCamera(stage.camera)
+    return { camera: this.camera, scene: this.scene }
+  }
+
+  private copyToLocalScene(scene: Scene) {
+    this.scene.copy(scene)
+    this.scene.children.length && this.scene.remove(...this.scene.children)
+    for (const child of scene.children) {
+      // Hack: Remove the parent so that three.js doesn't remove the children from the original scene object.
```

Comment:
Do we care? Won't the ones using this be recreating the scene object each time anyway?

Reply:
Try removing this line and see what happens 😛 (all the stories go blank)

Reply:
Seems important then! :P

Reply:
I agree that it should be regenerated 99% of the time, so I'll still look into why exactly this is needed.

...but I'll merge this for now 🙂